boilerplate/app/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,272 @@
+   */
+  labelTxOptions?: TextProps["txOptions"]
+  /**
+   * Style overrides for label text.
+   */
+  labelStyle?: StyleProp<TextStyle>
+  /**
+   * Pass any additional props directly to the label Text component.
+   */
+  LabelTextProps?: TextProps
+  /**
+   * The helper text to display if not using `helperTx`.
+   */
+  helper?: TextProps["text"]
+  /**
+   * Helper text which is looked up via i18n.
+   */
+  helperTx?: TextProps["tx"]
+  /**
+   * Optional helper options to pass to i18n. Useful for interpolation
+   * as well as explicitly setting locale or translation fallbacks.
+   */
+  helperTxOptions?: TextProps["txOptions"]
+  /**
+   * Pass any additional props directly to the helper Text component.
+   */
+  HelperTextProps?: TextProps
+  /**
+   * The input control for the type of toggle component
+   */
+  ToggleInput: FC<BaseToggleInputProps<T>>
+}
+
+export interface BaseToggleInputProps<T> {
+  on: boolean
+  status: ToggleProps<T>["status"]
+  disabled: boolean
+  outerStyle: ViewStyle
+  innerStyle: ViewStyle
+  detailStyle: Omit<ViewStyle & ImageStyle, "overflow">
+}
+
+/**
+ * Renders a boolean input.
+ * This is a controlled component that requires an onValueChange callback that updates the value prop in order for the component to reflect user actions. If the value prop is not updated, the component will continue to render the supplied value prop instead of the expected result of any user actions.
+ * @see [Documentation and Examples]{@link https://docs.infinite.red/ignite-cli/boilerplate/components/Toggle/}
+ * @param {ToggleProps} props - The props for the `Toggle` component.
+ * @returns {JSX.Element} The rendered `Toggle` component.
+ */
+export function Toggle<T>(props: ToggleProps<T>) {
+  const {
+    editable = true,
+    status,
+    value,
+    onPress,
+    onValueChange,
+    labelPosition = "right",
+    helper,
+    helperTx,
+    helperTxOptions,
+    HelperTextProps,
+    containerStyle: $containerStyleOverride,
+    inputWrapperStyle: $inputWrapperStyleOverride,
+    ToggleInput,
+    ...WrapperProps
+  } = props
+
+  const disabled = editable === false || status === "disabled" || props.disabled
+
+  const Wrapper = useMemo(
+    () => (disabled ? View : TouchableOpacity) as ComponentType<TouchableOpacityProps | ViewProps>,
+    [disabled],
+  )
+
+  const $containerStyles = [$containerStyleOverride]
+  const $inputWrapperStyles = [$inputWrapper, $inputWrapperStyleOverride]
+  const $helperStyles = [
+    $helper,
+    status === "error" && { color: colors.error },
+    HelperTextProps?.style,
+  ]
+
+  /**
+   * @param {GestureResponderEvent} e - The event object.
+   */
+  function handlePress(e: GestureResponderEvent) {
+    if (disabled) return
+    onValueChange?.(!value)
+    onPress?.(e)
+  }
+
+  return (
+    <Wrapper
+      activeOpacity={1}
+      // accessibilityRole={variant} // TODO: solve this since we no longer have variant
```

Comment:
We can't use displayName to determine this, can we we? Or would that be "ToggleInput". We could definitely just pass in `accessibilityRole` and that would get spread below.

Reply:
In the new revision, I passed it in from each of the parents with the variant names we would have had before - with the spread coming after in the event someone wanted to override it.